src/main/java/org/opensearch/knn/quantization/quantizer/OneBitScalarQuantizer.java

```diff
@@ -0,0 +1,104 @@
+import org.opensearch.knn.quantization.sampler.SamplingFactory;
+
+import java.io.IOException;
+import java.util.BitSet;
+
+/**
+ * OneBitScalarQuantizer is responsible for quantizing vectors using a single bit per dimension.
+ * It computes the mean of each dimension during training and then uses these means as thresholds
+ * for quantizing the vectors.
+ */
+public class OneBitScalarQuantizer implements Quantizer<float[], byte[]> {
+    private final int samplingSize; // Sampling size for training
+    private static final boolean IS_TRAINING_REQUIRED = true;
+    private final Sampler sampler; // Sampler for training
+    // Currently Lucene has sampling size as
+    // 25000 for segment level training , Keeping same
+    // to having consistent, Will revisit
+    // if this requires change
+    private static final int DEFAULT_SAMPLE_SIZE = 25000;
+
+    /**
+     * Constructs a OneBitScalarQuantizer with a default sampling size of 25000.
+     */
+    public OneBitScalarQuantizer() {
+        this(DEFAULT_SAMPLE_SIZE, SamplingFactory.getSampler(SamplerType.RESERVOIR));
+    }
+
+    /**
+     * Constructs a OneBitScalarQuantizer with a specified sampling size.
+     *
+     * @param samplingSize the number of samples to use for training.
+     */
+    public OneBitScalarQuantizer(final int samplingSize, final Sampler sampler) {
+
+        this.samplingSize = samplingSize;
+        this.sampler = sampler;
+    }
+
+    /**
+     * Trains the quantizer by calculating the mean of each dimension from the sampled vectors.
+     * These means are used as thresholds in the quantization process.
+     *
+     * @param trainingRequest the request containing the data and parameters for training.
+     * @return a OneBitScalarQuantizationState containing the calculated means.
+     */
+    @Override
+    public QuantizationState train(final TrainingRequest<float[]> trainingRequest) {
+        BitSet sampledDocIds = sampler.sample(trainingRequest.getTotalNumberOfVectors(), samplingSize);
+        float[] meanThresholds = QuantizerHelper.calculateMeanThresholds(trainingRequest, sampledDocIds);
+        return new OneBitScalarQuantizationState(new ScalarQuantizationParams(ScalarQuantizationType.ONE_BIT), meanThresholds);
+    }
+
+    /**
+     * Quantizes the provided vector using the given quantization state.
+     * It compares each dimension of the vector against the corresponding mean (threshold) to determine the quantized value.
+     *
+     * @param vector the vector to quantize.
+     * @param state  the quantization state containing the means for each dimension.
+     * @param output the QuantizationOutput object to store the quantized representation of the vector.
+     * @return a BinaryQuantizationOutput containing the quantized data.
+     */
+    @Override
+    public void quantize(final float[] vector, final QuantizationState state, final QuantizationOutput<byte[]> output) throws IOException {
```

Comment:
Why not make state a constructor parameter instead of passing in here?

Reply:
The reason why I am making like this is make Quantizer Stateless , so that we don't have to get quantizer per quantizer state.